src/main.rs

```diff
@@ -34,8 +65,41 @@ async fn main() {
         namespace: matches.value_of("cluster_namespace").unwrap_or_default(),
     };
 
-    let kub_controller = Kubernetes::new(environment_config.namespace).await;
+    let deploy_method = matches.value_of("deploy_method").unwrap();
+    let local_path = matches.value_of("local-path");
+    match deploy_method {
+        method if method == DeployMethod::Local.to_string() => {
+            if local_path.is_none() {
+                panic!("Error: --local-path is required for 'local' deploy-method.");
+            }
+        }
+        _ => {
+            if local_path.is_some() {
+                warn!("WARN: --local-path <path> will be ignored");
+            }
+        }
+    }
 
+    let solana_root = match matches.value_of("local-path") {
```

Comment:
nit: you already extracted this earlier, so it can just be `let solana_root = match local_path ...`